### ISSUE TITLE
Server does not send a response when a handler is not found

### ISSUE THREAD

**Issue by Bondza** [vmagamedov/grpclib#24]
Hello!
By accident I tried using `grpc_cli ls` when trying grpclib and I noticed that `grpc_cli` never exits and I think it's because it never gets a response.

The server outputs:
```
Server error
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/grpclib/server.py", line 139, in request_handler
    assert method is not None, h2_path
AssertionError: /grpc.reflection.v1alpha.ServerReflection/ServerReflectionInfo
```
So then it should run these lines of [code](https://github.com/vmagamedov/grpclib/blob/master/grpclib/server.py#L252-L255).

Should it not return `Status.UNIMPLEMENTED` in this case? It looks like this is what [grpc](https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/_server.py#L623-L625) does.

**Comment by vmagamedov** [comment]
Yep, you are right. Have to replace those assertions with proper errors handling and eliminate such situations with unclosed streams. Already started to fix this.

**Comment by vmagamedov** [comment]
Fixed in 7d144ee74e5117cc13e546a0b423598787de242b and b1c8b6de06c9419724f6f5c025c42fbd3632f860.
